subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/beans/BeanSchema.kt

```diff
@@ -31,24 +31,18 @@ import java.lang.reflect.Modifier
 
 internal
 fun relevantStateOf(taskType: Class<*>): Sequence<Field> =
-    relevantTypeHierarchyOf(taskType).flatMap { type ->
-        type.declaredFields.asSequence().filterNot { field ->
-            Modifier.isStatic(field.modifiers)
-                // Ignore the `metaClass` field that Groovy generates
-                || (field.isSynthetic && field.name == "metaClass" && MetaClass::class.java.isAssignableFrom(field.type))
-                // Ignore the `__meta_class__` field that Gradle generates
-                || (field.name == "__meta_class__" && MetaClass::class.java.isAssignableFrom(field.type))
-        }
-    }.onEach {
-        it.isAccessible = true
-    }
+    relevantTypeHierarchyOf(taskType)
+        .flatMap(Class<*>::relevantFields)
+        .map(Field::accessible)
```

Comment:
❌ `map` makes me think `Field::accessible` is returning a different value instead of acting on the given value, I'd stick to `onEach` (`.onEach(Field::makeAccessible)` maybe?)

Reply:
Fixed